src/components/CodeInput/inputItems.js

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { useReduxForm } from './../../libs/hooks';
+import { getLoopArray, getValidValue, isNumeric } from './helpers';
+import { StyledInput } from './styled';
+
+const setFocus = ref => {
+    if (ref && ref.current) {
+        ref.current.focus();
+        ref.current.select();
+    }
+};
+
+const InputItems = React.forwardRef((props, ref) => {
+    const {
+        value,
+        codeLength,
+        disabled,
+        required,
+        readOnly,
+        error,
+        tabIndex,
+        onClick,
+        onChange,
+        onFocus,
+        onBlur,
+        onKeyDown,
+    } = useReduxForm(props);
+
+    const arrayValue = value.split('');
+    const defaultActiveIndex = value ? arrayValue.length - 1 : 0;
+    const [activeIndex, setActiveIndex] = useState(defaultActiveIndex);
+
+    useEffect(() => {
+        setFocus(ref);
+    });
```

Comment:
this will cause the first input to get the focus and we need to defer this to the user, the consumer is responsible for focus the input whenever he wants